workers/unity/Packages/com.improbable.gdk.querybasedinteresthelper/Constraint.cs

```diff
@@ -0,0 +1,388 @@
+using System.Collections.Generic;
+using System.Linq;
+using Improbable.Gdk.Core;
+using UnityEngine;
+using UnityEngine.Assertions;
+
+namespace Improbable.Gdk.QueryBasedInterest
+{
+    /// <summary>
+    ///     Utility class to help construct QueryConstraint objects.
+    /// </summary>
+    public static class Constraint
+    {
+        private static readonly IReadOnlyList<ComponentInterest.QueryConstraint> EmptyList
+            = new List<ComponentInterest.QueryConstraint>();
+
+        private static ComponentInterest.QueryConstraint Default()
+        {
+            return new ComponentInterest.QueryConstraint
+            {
+                AndConstraint = EmptyList.ToList(),
+                OrConstraint = EmptyList.ToList()
+            };
+        }
+
+        /// <summary>
+        ///     Creates a QueryConstraint object with a Sphere constraint.
+        /// </summary>
+        /// <param name="radius">
+        ///     Radius of the Sphere constraint.
+        /// </param>
+        /// <param name="center">
+        ///     Center of the Sphere constraint.
+        /// </param>
+        /// <returns>
+        ///     A QueryConstraint object.
+        /// </returns>
+        public static ComponentInterest.QueryConstraint Sphere(double radius, Coordinates center)
+        {
+            var constraint = Default();
+            constraint.SphereConstraint = new ComponentInterest.SphereConstraint
+            {
+                Center = center,
+                Radius = radius
+            };
+            return constraint;
+        }
+
+        /// <summary>
+        ///     Creates a QueryConstraint object with a Sphere constraint.
+        /// </summary>
+        /// <param name="radius">
+        ///     Radius of the Sphere constraint.
+        /// </param>
+        /// <param name="centerX">
+        ///     X coordinate of the center of the Sphere constraint.
+        /// </param>
+        /// <param name="centerY">
+        ///     Y coordinate of the center of the Sphere constraint.
+        /// </param>
+        /// <param name="centerZ">
+        ///     Z coordinate of the center of the Sphere constraint.
+        /// </param>
+        /// <returns>
+        ///     A QueryConstraint object.
+        /// </returns>
+        public static ComponentInterest.QueryConstraint Sphere(
```

Comment:
Have this call into the Coordinates version instead to avoid code duplication

Reply:
Agreed. Or the other way around. There is an interesting question around how this will mesh with coordinate remapping though. Doesn't need to be solved now but it is worth thinking about.